### PR TITLE
Move jenkins/ dir into ci/jenkins and spread docs around

### PR DESCRIPTION
**Pull request by areusch**
Reorganizing some of the Jenkins docs to make it easier to present.

* `jenkins/README.md` -> `ci/README.md` and retain the overview section
* Move the infrastructure related parts to `tlc-pack/ci` so that they live along with that impl.

@driazati @konturn

### CODE REVIEW

File: ci/jenkins/README.md [hunk 1]
```diff
@@ -33,17 +37,20 @@ https://github.com/apache/tvm/actions has the logs for each of these workflows.
...
+for the author of the offending PR when that PR is large.
 
-Some tests are also flaky and fail for reasons unrelated to the PR. The [CI monitoring rotation](https://github.com/apache/tvm/wiki/CI-Monitoring-Runbook) watches for these failures and disables tests as necessary. It is the responsibility of those who wrote the test to ultimately fix and re-enable the test.
+Some tests are also flaky and occasionally fail for reasons unrelated to the PR. The [CI monitoring
+rotation](https://github.com/apache/tvm/wiki/CI-Monitoring-Runbook) watches for these failures and
+disables tests as necessary. It is the responsibility of those who wrote the test to ultimately fix
+and re-enable the test.
 
 
 ## Dealing with Flakiness
```
Comment (driazati) [inline ci/jenkins/README.md:56]: This stuff should be either in .rst or in .md, can you delete one or the other?

File: ci/jenkins/README.md [hunk 2]
```diff
@@ -153,6 +160,7 @@ _venv/bin/python3 jenkins/generate.py
 
 # Infrastructure
 
+While all TVM tests are contained within the apache/tvm repository, the infrastructure used to run the tests is donated by the TVM Community. To encourage collaboration,
 Jenkins runs in AWS on an EC2 instance fronted by an ELB which makes it available at https://ci.tlcpack.ai. These definitions are declared via Terraform in the [tlc-pack/ci-terraform](https://github.com/tlc-pack/ci-terraform) repository. The Terraform code references custom AMIs built in [tlc-pack/ci-packer](https://github.com/tlc-pack/ci-packer). [tlc-pack/ci](https://github.com/tlc-pack/ci) contains Ansible scripts to deploy the Jenkins head node and set it up to interact with AWS.
 
 The Jenkins head node has a number of autoscaling groups with labels that are used to run jobs (e.g. `CPU`, `GPU` or `ARM`) via the [EC2 Fleet](https://plugins.jenkins.io/ec2-fleet/) plugin.
```
Comment (driazati) [inline ci/jenkins/README.md:166]: the tlcpack stuff is merged now, can you delete the diagram from here?

**Review by driazati — APPROVED** [review]

File: ci/jenkins/README.md [hunk 3]
```diff
@@ -153,6 +160,7 @@ _venv/bin/python3 jenkins/generate.py
 
 # Infrastructure
 
+While all TVM tests are contained within the apache/tvm repository, the infrastructure used to run the tests is donated by the TVM Community. To encourage collaboration,
```
[the remark this answers is not in the document]
Comment (areusch) [reply ci/jenkins/README.md:163]: oops, fixed

File: ci/jenkins/README.md (hunk 2 above)
Comment (areusch) [reply ci/jenkins/README.md:166]: done

File: ci/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+1. User-facing documentation lives in `apache/tvm`'s `docs/contribute` sub-directory and is served on the
+   [TVM docs site](https://tvm.apache.org/docs/contribute/ci.html).
+2. Documentation of the tools that run TVM's various regression tests locally and the test suites
+   are in this sub-directory.
+3. Documentation of the cloud services and their configuration lives in the
+   [`tlc-pack/ci`](https://github.com/tlc-pack/ci) repo.
+
+## Jenkins
+
+Jenkins runs all of the linux-based TVM CI-enabled regression tests. This includes tests against accelerated hardware such as GPUs. It excludes those regression tests that run against hardware not available in the cloud (those tests aren't currently exercised in TVM CI). The tests run by Jenkins represent most of the merge-blocking tests (and passing Jenkins should mostly correlate with passing the remaining Windows/Mac builds).
```
[the remark this answers is not in the document]
Comment (areusch) [reply ci/README.md:51]: done

File: docs/contribute/ci.rst [hunk 5]
```diff
@@ -23,14 +23,21 @@ Using TVM's CI
...
+TVM primarily uses Jenkins for running Linux continuous integration (CI) tests on
+`branches <https://ci.tlcpack.ai/job/tvm/>`_
 `pull requests <https://ci.tlcpack.ai/job/tvm/view/change-requests/>`_ through a
 build configuration specified in a `Jenkinsfile <https://github.com/apache/tvm/blob/main/Jenkinsfile>`_.
-Non-critical jobs run in GitHub Actions for Windows and MacOS jobs.
+Jenkins is the only CI step that is codified to block merging. TVM is also tested minimally
+against Windows and MacOS using GitHub Actions.
+
+This page describes how contributors and committers can use TVM's CI to verify their code. You can
+read more about the design of TVM CI in the
```
[the remark this answers is not in the document]
Comment (areusch) [reply docs/contribute/ci.rst:34]: done

File: ci/jenkins/README.md (hunk 1 above)
Comment (areusch) [reply ci/jenkins/README.md:56]: oh whoops, this was pretty sloppy. done.

**Pull request merged**